### ISSUE TITLE
WebSocket Peers are not being deleted

### ISSUE THREAD

**Issue by DiegoRBaquero** [feross/bittorrent-tracker#133]
Version 7.5.4

For the moment, that's all the information I have.

**Comment by DiegoRBaquero** [comment]
The socket's infoHashes is empty for every peer in my tracker's debug.

I can only seem to point it to this change: https://github.com/feross/bittorrent-tracker/commit/39507bf8be3f391abef7c2258b32b6fe5cfcaa27#diff-78c12f5adc1848d13b1c6f07055d996eL339

Will keep digging into it.

**Comment by yciabaud** [comment]
In this change I am converting single hash to an array:
```
if (typeof params.info_hash === 'string') hashes = [ params.info_hash ]
else hashes = params.info_hash
```
And then I am iterating over this array to add it to the socket : 
```
hashes.forEach(function (info_hash) {
      if (socket.infoHashes.indexOf(info_hash) === -1) {
        socket.infoHashes.push(info_hash)
      }
})
```

Do you think I missed something?

**Comment by DiegoRBaquero** [comment]
Feross removed it here: https://github.com/feross/bittorrent-tracker/commit/ba4a52d2d14eb81ccab291a7ed124d051829dfbf#diff-78c12f5adc1848d13b1c6f07055d996eL345

Now I'm looking on how to fix it

**Comment by DiegoRBaquero** [comment]
@yciabaud PR sent bringing back your code.

**Comment by feross** [comment]
My bad – I wanted to move the code that adds to the infoHashes array to inside the `if (response.action === 'announce')` block because it should not happen for scrape. But looks like it got removed entirely!

**Comment by feross** [comment]
7.5.5

**Comment by DiegoRBaquero** [comment]
@feross Still not getting deleted and now erroring:

`btorrent-tracker-0 Error: not opened
btorrent-tracker-0     at WebSocket.send (/root/tracker/source/node_modules/bittorrent-tracker/node_modules/ws/lib/WebSocket.js:218:16)
btorrent-tracker-0     at /root/tracker/source/node_modules/bittorrent-tracker/server.js:370:21
btorrent-tracker-0     at Array.forEach (native)
btorrent-tracker-0     at /root/tracker/source/node_modules/bittorrent-tracker/server.js:369:13
btorrent-tracker-0     at /root/tracker/source/node_modules/bittorrent-tracker/server.js:497:7
btorrent-tracker-0     at Swarm.announce (/root/tracker/source/node_modules/bittorrent-tracker/lib/server/swarm.js:31:3)
btorrent-tracker-0     at announce (/root/tracker/source/node_modules/bittorrent-tracker/server.js:465:11)
btorrent-tracker-0     at /root/tracker/source/node_modules/bittorrent-tracker/server.js:435:7
btorrent-tracker-0     at /root/tracker/source/node_modules/bittorrent-tracker/server.js:214:5
btorrent-tracker-0     at nextTickCallbackWith0Args (node.js:420:9)`

**Comment by feross** [comment]
Ugh. My bad. We need more tests for the websocket server.

Should be fixed in 7.5.7
